### PR TITLE
metrics, telemetry: add telemetry for metadata lock

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: ref #37275

Problem Summary:
add telemetry for metadata lock

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: metrics/telemetry.go [hunk 1]
```diff
@@ -328,6 +328,7 @@ func GetLazyPessimisticUniqueCheckSetCounter() int64 {
 // DDLUsageCounter records the usages of Add Index with acceleration solution.
 type DDLUsageCounter struct {
 	AddIndexIngestUsed int64 `json:"add_index_Ingest_used"`
+	MetadataLockUsed   bool  `json:"metadata_lock_used"`
```
Comment (Benjamin2037) [inline metrics/telemetry.go:331]: Do you not want to know how many user enable/disable MDL?

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by AstroProfundis — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Comment by hawkingrei** [comment]
@wjhuang2016 Please check the unit test.

File: metrics/telemetry.go [hunk 2]
```diff
@@ -328,6 +328,7 @@ func GetLazyPessimisticUniqueCheckSetCounter() int64 {
 // DDLUsageCounter records the usages of Add Index with acceleration solution.
 type DDLUsageCounter struct {
 	AddIndexIngestUsed int64 `json:"add_index_Ingest_used"`
```
Comment (tangenta) [inline metrics/telemetry.go:330]:
```suggestion
	AddIndexIngestUsed int64 `json:"add_index_ingest_used"`
```

**Pull request merged**